policies/policy_service.go

```diff
@@ -218,4 +219,32 @@ func (s policiesService) ValidatePolicy(ctx context.Context, token string, p Pol
 	p.MFOwnerID = mfOwnerID
 
 	return p, nil
+}
+func (s policiesService) ValidateDataset(ctx context.Context, token string, d Dataset) (Dataset, error) {
+	mfOwnerID, err := s.identify(token)
+	if err != nil {
+		return Dataset{}, err
+	}
+
+	d.MFOwnerID = mfOwnerID
+
+	_, err = uuid.FromString(d.SinkID)
+	if err != nil {
+		fmt.Println("invalid sink id")
+		return Dataset{}, ErrMalformedEntity
+	}
+
```

Comment:
In addition to making sure that the ID is a valid UUID, we also want to make sure that the ID represents a valid Sink. This means it should be making a gRPC call to ensure it can retrieve a Sink with that ID for this owner. The same goes for policy and group ID.